packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -115,6 +116,7 @@ function BlockInspectorLockedBlocks( { topLevelLockedBlock } ) {
 				className={ blockInformation.isSynced && 'is-synced' }
 			/>
 			<BlockVariationTransforms blockClientId={ topLevelLockedBlock } />
+			<BlockInfo.Slot />
```

Comment:
Shouldn't  this be above `BlockVariationTransforms`? Also related [issue](https://github.com/WordPress/gutenberg/issues/49887) about slot in BlockCard.

Reply:
When the block inspector tabs were being introduced, we explored adding a new slot above the tabs for items that related to the block as a whole but weren't really a setting or style. At that time the direction was that it should occur the block transforms as those should still have a higher prominence.

In the Query block's use case with the "add new post" link, it doesn't particularly matter. 

It's worth noting that the earlier exploration into adding a slot above the block inspector tabs was blocked due to concerns about the slot being abused by 3rd parties that might promote their controls in ways detrimental to the end-user experience. The potential "resistance" to a new slot fill here was questioned again in earlier comments on this PR. It was these concerns that drove the approach in this PR to make this proposed slot "private". 

A "private" slot as proposed here wouldn't solve the need expressed in https://github.com/WordPress/gutenberg/issues/49887 as the use case there requires it to be available for public consumption.

So that leaves me a little unsure of how best to proceed here. Some options are:

1. Proceed with this private slot approach to solve the Query block's immediate need.
    - Buys time to get consensus on if we are in fact happy to add a public slot
    - Allows time for exploration to see if we might mitigate misuse of the public slot by filtering allowed fills or something
    - It can easily be changed or removed should we land a new public slot and the private slot fills might be beneficial elsewhere.
2. Shelve this fix until we get consensus on a new public slot, including what is acceptable to be added above the block transforms
3. Bite the bullet and add a public slot to the BlockCard now
4. Rework this PR to only add the private SlotFill feature omitting its use in the block inspector and query block for now

While I don't feel strongly, I'm leaning more towards option 1, landing this PR and iterating from there. That leaning was reinforced further after some internal discussions with @talldan.

What do you think? Are there any better options I haven't considered?

Reply:
Sounds good, thank you Aaron!